depthai_demo.py

```diff
@@ -171,7 +218,35 @@ def get_label_text(self, label):
             return self.labels[int(label)]
         else:
             print(f"Label of ouf bounds (label_index: {label}, available_labels: {len(self.labels)}")
-            return label
+            return str(label)
+    
+    def draw_detections(self, source, detections):
+        def draw_detection(frame, detection):
+            bbox = frame_norm(frame, [detection.xmin, detection.ymin, detection.xmax, detection.ymax])
+            cv2.rectangle(frame, (bbox[0], bbox[1]), (bbox[2], bbox[3]), (255, 0, 0), 2)
+            cv2.putText(frame, self.get_label_text(detection.label), (bbox[0] + 10, bbox[1] + 20),
+                        cv2.FONT_HERSHEY_TRIPLEX, 0.5, 255)
+            cv2.putText(frame, f"{int(detection.confidence * 100)}%", (bbox[0] + 10, bbox[1] + 40),
+                        cv2.FONT_HERSHEY_TRIPLEX, 0.5, 255)
+
+            if conf.useDepth:  # Display coordinates as well
+                cv2.putText(frame, f"X: {int(detection.spatialCoordinates.x)} mm", (bbox[0] + 10, bbox[1] + 60),
+                            cv2.FONT_HERSHEY_TRIPLEX, 0.5, color)
+                cv2.putText(frame, f"Y: {int(detection.spatialCoordinates.y)} mm", (bbox[0] + 10, bbox[1] + 75),
+                            cv2.FONT_HERSHEY_TRIPLEX, 0.5, color)
+                cv2.putText(frame, f"Z: {int(detection.spatialCoordinates.z)} mm", (bbox[0] + 10, bbox[1] + 90),
+                            cv2.FONT_HERSHEY_TRIPLEX, 0.5, color)
```

Comment:
Rename `color` to `text_color` to be more explicit and easily configurable, if one wants to. (No need to add param)